src/shared/components/tutor-personal-modal/tutor-personal-modal.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, Inject, OnInit } from '@angular/core'; 
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { SubjectService } from 'src/shared/providers/subject.service';
+import { TutorService } from 'src/shared/providers/tutor.service';
+import { AuthService } from 'src/shared/providers/auth.service';
+import { ISubjects, ITutorPersonalModal } from 'src/shared/interfaces/ITutorPersonalData.interface';
+import { LoadingService } from 'src/shared/providers/loading.service';
+
+@Component({
+  selector: 'app-tutor-personal-modal',
+  templateUrl: './tutor-personal-modal.component.html',
+  styleUrls: ['./tutor-personal-modal.component.scss']
+})
+export class TutorPersonalModalComponent implements OnInit {
+  subjects: string[] = [];
+  tempSelectedSubjects: string[] = [];
+  username: string = '';
```

Comment:
Se for usar o formgroup, pra usar os reactive forms precisa mudar as estruturas aqui